src/plugins/dashboard/public/services/dashboard_content_management/lib/find_dashboards.ts

```diff
@@ -83,19 +83,25 @@ export async function findDashboardById(
       references: cachedDashboard.item.references,
     };
   }
+
   /** Otherwise, fetch the dashboard from the content management client, add it to the cache, and return the result */
-  const response = await contentManagement.client
-    .get<DashboardCrudTypes['GetIn'], DashboardCrudTypes['GetOut']>({
+  try {
+    const response = await contentManagement.client.get<
+      DashboardCrudTypes['GetIn'],
+      DashboardCrudTypes['GetOut']
+    >({
       contentTypeId: DASHBOARD_CONTENT_ID,
       id,
-    })
-    .then((result) => {
-      dashboardContentManagementCache.addDashboard(result);
-      return { id, status: 'success', attributes: result.item.attributes };
-    })
-    .catch((e) => ({ status: 'error', error: e.body, id }));
-
-  return response as FindDashboardsByIdResponse;
+    });
+    dashboardContentManagementCache.addDashboard(response);
+    return {
+      id,
+      status: 'success',
+      attributes: response.item.attributes,
+    } as FindDashboardsByIdResponse;
+  } catch (e) {
+    return { status: 'error', error: e.body || e.message, id } as FindDashboardsByIdResponse;
+  }
```

Comment:
Old method wasn't catching errors being thrown in `contentManagement.client.get`